From d61eff99a44732ccc2e6e46cf895f9a27efbeb4b Mon Sep 17 00:00:00 2001
From: sapphi-red <49056869+sapphi-red@users.noreply.github.com>
Date: Wed, 26 Mar 2025 22:27:14 +0900
Subject: [PATCH] temporary commit

---
 packages/plugin-vue/src/index.ts | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/packages/plugin-vue/src/index.ts b/packages/plugin-vue/src/index.ts
index 4606b8b..ba6517d 100644
--- a/packages/plugin-vue/src/index.ts
+++ b/packages/plugin-vue/src/index.ts
@@ -314,9 +314,11 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin<Api> {
       transformCachedModule =
         config.command === 'build' &&
         options.value.sourceMap &&
+        // @ts-expect-error rolldown-vite does not support watch mode yet
         config.build.watch != null
     },
 
+    // @ts-expect-error allow any
     shouldTransformCachedModule({ id }) {
       if (transformCachedModule && parseVueRequest(id).query.vue) {
         return true
@@ -389,6 +391,7 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin<Api> {
       }
     },
 
+    // @ts-expect-error rolldown type incompat
     transform(code, id, opt) {
       const ssr = opt?.ssr === true
       const { filename, query } = parseVueRequest(id)
@@ -407,6 +410,7 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin<Api> {
           code,
           filename,
           options.value,
+          // @ts-expect-error rolldown type incompat
           this,
           ssr,
           customElementFilter.value(filename),
@@ -427,6 +431,7 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin<Api> {
             code,
             descriptor,
             options.value,
+            // @ts-expect-error rolldown type incompat
             this,
             ssr,
             customElementFilter.value(filename),
@@ -437,6 +442,7 @@ export default function vuePlugin(rawOptions: Options = {}): Plugin<Api> {
             descriptor,
             Number(query.index || 0),
             options.value,
+            // @ts-expect-error rolldown type incompat
             this,
             filename,
           )
-- 
2.48.1

