From 3db74d3db8d0c21030209d7a3bbd52ebd1387ab2 Mon Sep 17 00:00:00 2001
From: sapphi-red <49056869+sapphi-red@users.noreply.github.com>
Date: Thu, 27 Mar 2025 16:41:19 +0900
Subject: [PATCH] temporary commit

---
 src/index.ts | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/index.ts b/src/index.ts
index c48f8f8..b15ad6a 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -99,6 +99,7 @@ const react = (_options?: Options): PluginOption[] => {
       apply: "serve",
       config: () => ({
         esbuild: false,
+        oxc: false,
         optimizeDeps: {
           include: [`${options.jsxImportSource}/jsx-dev-runtime`],
           esbuildOptions: { jsx: "automatic" },
-- 
2.48.1

